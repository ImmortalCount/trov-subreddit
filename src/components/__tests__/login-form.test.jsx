/* eslint-disable no-magic-numbers */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import LoginForm from '../login-form'

describe('LoginForm Component', () => {
    const defaultProps = {
        onValidSubmit: jest.fn(),
        handleLoginSubmit: jest.fn(),
        clearError: jest.fn()
    }
    const wrapper = shallow(
        <LoginForm {...defaultProps} />
    )

    it('should render LoginForm component', () => {
        expect(wrapper.find('Message').length).toBe(1)
        expect(wrapper.find('Button').length).toBe(1)
    })

    it('should display error message', () => {
        const isError = { isError: true }
        const newDefaultProps = { ...defaultProps, ...isError }
        const loginErrorRender = renderer.create(
            <LoginForm {...newDefaultProps} />
        ).toJSON()
        expect(loginErrorRender).toMatchSnapshot()
    })

    it('should fire form handler', () => {
        wrapper.find('Form').props().onValidSubmit()
        expect(defaultProps.handleLoginSubmit).toBeCalled()
    })

    it('LoginForm matches snapshot', () => {
        const loginFormRender = renderer.create(
            <LoginForm {...defaultProps} />
        ).toJSON()
        expect(loginFormRender).toMatchSnapshot()
    })
})
