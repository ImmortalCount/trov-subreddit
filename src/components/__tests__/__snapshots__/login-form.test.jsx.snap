// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component LoginForm matches snapshot 1`] = `
<div
  className="ui message login-form"
>
  <form
    disabled={false}
    noValidate={true}
    onError={[Function]}
    onFocus={[MockFunction]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ui form"
      onFocus={[MockFunction]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="username"
        >
           
          Email Address
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          htmlFor="password"
        >
           
          Password
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        className="ui green button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`LoginForm Component should display error message 1`] = `
<div
  className="ui message login-form"
>
  <div
    className="ui red icon message"
  >
    <i
      aria-hidden="true"
      className="warning sign icon"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Error
      </div>
      <p>
        There was an error logging in.
      </p>
    </div>
  </div>
  <form
    disabled={false}
    noValidate={true}
    onError={[Function]}
    onFocus={[MockFunction]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ui form"
      onFocus={[MockFunction]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="username"
        >
           
          Email Address
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          htmlFor="password"
        >
           
          Password
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        className="ui green button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
